accounts/keystore/key2335.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2014 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// Modified by Prysmatic Labs 2018
+// Modified by the klaytn Authors 2023
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package keystore
+
+import (
+	"encoding/hex"
+	"encoding/json"
+
+	"github.com/klaytn/klaytn/crypto/bls"
+	"github.com/pborman/uuid"
+	keystorev4 "github.com/wealdtech/go-eth2-wallet-encryptor-keystorev4"
+)
+
+// KeyEIP2335 is a decrypted BLS12-381 keypair.
+type KeyEIP2335 struct {
+	ID uuid.UUID // Version 4 "random" for unique id not derived from key data
+
+	PublicKey bls.PublicKey // Represents the public key of the user.
+	SecretKey bls.SecretKey // Represents the private key of the user.
+}
+
+type encryptedKeyEIP2335JSON struct {
+	PublicKey string                 `json:"publickey"`
+	Crypto    map[string]interface{} `json:"crypto"`
+	ID        string                 `json:"id"`
+}
+
+// NewKeyEIP2335 creates a new EIP-2335 keystore Key type using a BLS private key.
+func NewKeyEIP2335(blsKey bls.SecretKey) *KeyEIP2335 {
+	return &KeyEIP2335{
+		ID:        uuid.NewRandom(),
+		PublicKey: blsKey.PublicKey(),
+		SecretKey: blsKey,
+	}
+}
+
+// DecryptKeyEIP2335 decrypts a key from an EIP-2335 JSON blob, returning the BLS private key.
+func DecryptKeyEIP2335(keyJSON []byte, password string) (*KeyEIP2335, error) {
+	k := new(encryptedKeyEIP2335JSON)
+	if err := json.Unmarshal(keyJSON, k); err != nil {
+		return nil, err
+	}
+
+	decryptor := keystorev4.New()
+	keyBytes, err := decryptor.Decrypt(k.Crypto, password)
+	if err != nil {
+		return nil, err
+	}
+
+	secretKey, err := bls.SecretKeyFromBytes(keyBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KeyEIP2335{
```

Comment:
Seems to lack of parsing check for `ID`(specified as `uuid` in the provided test file).  The provided unit test also lacks this check.